quantum/process_keycode/process_repeat_key.c

```diff
@@ -0,0 +1,83 @@
+// Copyright 2022-2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "process_repeat_key.h"
+
+bool process_repeat_key(uint16_t keycode, keyrecord_t* record) {
+    if (get_repeat_key_count()) {
+        return true;
+    }
+
+    if (keycode == QK_REPEAT_KEY) {
+        repeat_key_invoke(&record->event);
+        return false;
+#ifndef NO_ALT_REPEAT_KEY
+    } else if (keycode == QK_ALT_REPEAT_KEY) {
+        alt_repeat_key_invoke(&record->event);
+        return false;
+#endif // NO_ALT_REPEAT_KEY
+    } else if (record->event.pressed && get_repeat_key_eligible(keycode, record)) {
+        set_repeat_key_record(keycode, record);
+        set_repeat_key_mods(get_mods() | get_weak_mods()
+#ifndef NO_ACTION_ONESHOT
+                            | get_oneshot_mods()
+#endif // NO_ACTION_ONESHOT
+        );
+    }
+
+    return true;
+}
+
+// Default implementation of get_repeat_key_eligible().
+__attribute__((weak)) bool get_repeat_key_eligible(uint16_t keycode, keyrecord_t* record) {
+    switch (keycode) {
+        // Ignore MO, TO, TG, and TT layer switch keys.
+        case QK_MOMENTARY ... QK_MOMENTARY_MAX:
```

Comment:
```suggestion
// Default implementation of get_repeat_key_eligible().
__attribute__((weak)) bool get_repeat_key_eligible(uint16_t keycode, keyrecord_t* record) {
    switch (keycode) {
        // Ignore MO, TO, TG, TT, and TL layer switch keys.
        case QK_TRI_LAYER_LOWER:
        case QK_TRI_LAYER_UPPER:
        case QK_MOMENTARY ... QK_MOMENTARY_MAX:
```

@getreuer I'm currently using the userspace implementation of your Repeat key, not the PR, but I feel like the Tri Layer specific layer keycodes should also be included in the default `get_repeat_key_eligible`, since they're essentially MO keycodes with additional function?

Reply:
Great catch! Thanks for the edit. Committed.